packages/clay-language-picker/src/index.tsx

```diff
@@ -0,0 +1,291 @@
+
+type Translations = {
+	[key: string]: Translation;
+};
+
+type Props = {
+	/**
+	 * Flag to indicate if the Picker is active or not (controlled).
+	 */
+	active?: boolean;
+
+	/**
+	 * Class names for the trigger.
+	 */
+
+	classNamesTrigger?: string;
+
+	/**
+	 * Flag to hide the text in the trigger.
+	 */
+
+	hideTriggerText?: boolean;
+
+	/**
+	 * The id of the component.
+	 */
+	id?: string;
+
+	/**
+	 * Labels for the component
+	 */
+	labels?: Labels;
+
+	/**
+	 * List of locales to allow localization for
+	 */
+	locales: Array<Item>;
+
+	/**
+	 * Callback for when the active state changes (controlled).
+	 */
+
+	onActiveChange?: InternalDispatch<boolean>;
+
+	/**
+	 * Callback that gets called when a selected locale gets changed
+	 */
+	onSelectedLocaleChange: (val: React.Key) => void;
+
+	/**
+	 * Exposes the currently selected locale
+	 */
+	selectedLocale: Item;
+
+	/**
+	 * Flag to show the picker in small size.
+	 */
+	small?: boolean;
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap?: string;
+
+	/**
+	 * Translations provided to the component to be used and modified by it
+	 */
+	translations?: Translations;
+};
+
+const TranslationLabel = ({
+	defaultLanguage,
+	labels,
+	locale,
+	translation,
+}: {
+	defaultLanguage: Item;
+	labels: Labels;
+	locale: Item;
+	translation: Translation;
+}) => {
+	let displayType: DisplayType = 'warning';
+	let label = labels.untranslated;
+
+	if (locale.label === defaultLanguage?.label) {
+		displayType = 'info';
+		label = labels.default;
+	} else if (translation) {
+		const {total, translated} = translation;
+
+		if (total && total === translated) {
+			displayType = 'success';
+			label = labels.translated;
+		} else {
+			displayType = 'secondary';
+			label = sub(labels.translating, [translated, total]);
+		}
+	}
+
+	return (
+		<>
+			<span className="sr-only">
+				{sub(labels.option, [
+					locale.displayName || locale.label,
+					label,
+				])}
+			</span>
+
+			<ClayLabel
+				aria-hidden="true"
+				className="c-ml-3"
+				displayType={displayType}
+			>
+				{label}
+			</ClayLabel>
+		</>
+	);
+};
+
+TranslationLabel.displayName = 'Label';
+
+const Trigger = React.forwardRef<HTMLButtonElement>(
+	(
+		{
+			ariaLabelTrigger,
+			classNamesTrigger,
+			hideTriggerText,
+			selectedLocale,
+			small,
+			spritemap,
+			...otherProps
+		}: Record<string, any>,
+		ref
+	) => {
+		return (
+			<ClayButton
+				{...otherProps}
+				aria-label={ariaLabelTrigger}
+				className={classNames(
+					classNamesTrigger,
+					'form-control-select',
+					{
+						'hidden-label': hideTriggerText,
+					}
+				)}
+				displayType="secondary"
+				ref={ref}
+				size={small ? 'sm' : undefined}
+			>
+				<span className="inline-item-before">
+					<ClayIcon
+						spritemap={spritemap}
+						symbol={selectedLocale.symbol}
+					/>
+				</span>
+
+				{!hideTriggerText ? (
+					<span className="font-weight-normal mr-2">
```

Comment:
Without the `btn` class we won't need `font-weight-normal`. Do we need `mr-2` here because the spacing is incorrect between the arrows and text? I can update in CSS if so.

Reply:
Yes, that margin is a safety spacing between the text and the arrows. Updating it in CSS would be ideal.
This change would only be for the case where the text is displayed, since the spacing is correct when only the flag is displayed.

<img width="176" alt="Screenshot 2025-01-17 at 4 02 09 PM" src="https://github.com/user-attachments/assets/9bcc0030-97b0-47b0-a8ca-4672c5ddb5a7" />